### ISSUE TITLE
For `PIN code input` component, add label on each digit and do not move focus

### ISSUE THREAD

**Issue by pya35** [Orange-OpenSource/ouds-ios#1409]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-ios/issues) for duplicate or closed feature requests

### Your test device

iPhone 14+

### Device OS version

26

### OUDS iOS library version

1.4.0

### Describe the issue

Digits don't have any accessible label
Focus moves automatically after a digit is entered 

### Expected Behavior

Each digit should have an individual label like "1st digit, 2nd digit..."

Group the digits to give a general context like "Entrez le code à 6 chiffres"

Note : Once a digit is entered, the focus should not move automatically to the next field

### Accessibility settings

VO

**Comment by pylapp** [comment]
@pya35 Fonctionnellement le focus doit être mis à la case suivante à chaque saisie, ce qui améliore l'UX.

Mais je comprends parfaitement ton point de vue ; du coup est-ce que l'on ne pourrait pas trouver un compromis par exemple en désactivant le focus automatique selon certains réglages d'accessibilité ?

**Comment by pya35** [comment]
> [@pya35](https://github.com/pya35) Fonctionnellement le focus doit être mis à la cause suivante à chaque saisie, ce qui améliorer l'UX.
> 
> Mais je comprends parfaitement ton point de vue ; du coup est-ce que l'on ne pourrait pas trouver un compromis par exemple en désactivant le focus automatique selon certains réglages d'accessibilité ?

@pylapp Oui c'est le fonctionnement "classique" de ce genre de composant, mais cela va à l'encontre du critère [WCAG 3.2.2 On Input](https://w3.org/TR/WCAG22#on-input) qui demande à ce que le focus ne soit pas déplacé automatiquement, sauf avertissement de l'utilisateur. C'est principalement lié aux utilisateurs mal/non-voyants qui doivent être maitres de la navigation, on pourrait donc imaginer de bloquer le focus si zoom ou VO activé ? mais ça impacte aussi les déficients cognitifs qui peuvent être perturbés et là pas d'aides techniques associées.

**Comment by pylapp** [comment]
@pya35 On verra ce que l'on pourra faire en fonction de ce que nous permet l'API Apple.

Ce que je propose : 
- Suivre tes recommendations sur la vocalisation 
- Suivre tes recommandations sur les labels des champs de saisie
- Voir si le focus automatique peut être désactivé si Voice Over est activé ou en cas de gros texte, genre dès 235% par exemple
- On met de côté sans oublier le sujet du cas des utilisateurs déficients cognitifs et de l'autofocus

Est-ce que ça te paraît acceptable ?

**Comment by pylapp** [comment]
> [!NOTE]
> Corrections faites : 
> - Label pour le composant
> - Label pour chaque champs
> - Désactivation de l'autofocus si Voice Over activé

**Comment by pylapp** [comment]
📣 **New TestFlight BETA upload available** 🚀

Please, get the new build from *TestFlight* app, or contact the project maintainers (pylapp, ludovic35) to get access and test this issue implementation.

- Display name: **Design System Toolbox (BETA)**
- Version: **1.4.0**
- Build number: **1776287078**
- Build tag: **6355b9e**


_🤖 This is an automated message sent with love thanks to *Fastlane* from our CI/CD pipeline 🤘_

**Comment by pya35** [comment]
@pylapp After doing some tests with both ways of navigation on Flutter, I'm agree that it's quite difficult to fill in the PIN code when focus is not moving automatically. The best solution is to move the focus on next input automatically as done previously.

But there's several issues for this case (tested on build 704) 
- when focus moves to the next field, the new focused input is not vocalized: this field shall be vocalized to inform the user
- when user deletes an input, ensure that focus stays in the deleted input
- inputs should be named like "1er chiffre, 2eme chiffre..." / "1st digit, 2nd digit...." 
- group inputs and name it like "entrez les 6 chiffres"
